cmd/compose/config.go

```diff
@@ -123,18 +135,56 @@ func configCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service
 	return cmd
 }
 
-func runConfig(ctx context.Context, dockerCli command.Cli, backend api.Service, opts configOptions, services []string) error {
+func runConfig(ctx context.Context, dockerCli command.Cli, opts configOptions, services []string) error {
 	var content []byte
-	project, _, err := opts.ToProject(ctx, dockerCli, services)
+	model, err := opts.ToModel(ctx, dockerCli, services)
 	if err != nil {
 		return err
 	}
 
-	content, err = backend.Config(ctx, project, api.ConfigOptions{
-		Format:              opts.Format,
-		Output:              opts.Output,
-		ResolveImageDigests: opts.resolveImageDigests,
-	})
+	if opts.resolveImageDigests {
+		// create a pseudo-project so we can rely on WithImagesResolved to resolve images
+		p := &types.Project{
+			Services: types.Services{},
+		}
+		services := model["services"].(map[string]any)
+		for name, s := range services {
+			service := s.(map[string]any)
+			if image, ok := service["image"]; ok {
+				p.Services[name] = types.ServiceConfig{
+					Image: image.(string),
+				}
+			}
+		}
+
+		p, err = p.WithImagesResolved(compose.ImageDigestResolver(ctx, dockerCli.ConfigFile(), dockerCli.Client()))
+		if err != nil {
+			return err
+		}
+
+		for name, s := range services {
+			service := s.(map[string]any)
+			config := p.Services[name]
+			if config.Image != "" {
+				service["image"] = config.Image
+			}
```

Comment:
we could add this in the previous loop

Reply:
no, because this is computed by `p.WithImagesResolved`: value isn't set to the expected value (image id with digest) before we call this func